db/src/Pos/DB/Block/Epoch.hs

```diff
@@ -198,50 +253,112 @@ data SlotIndexLength
-        (epochPath, indexPath) <- mkEpochPaths (ccpEpochIndex ccp) . view epochDataDir <$> getNodeDBs
-        liftIO $ print (epochPath, indexPath)
-
-        xs <- consolidateEpochBlocks epochPath sihs
-        liftIO $ writeEpochIndex epochSlots indexPath xs
-
-        -- Write starting point for next consolidation to the MiscDB.
-        let nextCcp = ConsolidateCheckPoint (ccpEpochIndex ccp + 1) epochBoundary
-        putConsolidateCheckPoint nextCcp
-
-        -- After the check point is written, delete old Blunds.
-        lift $ mapM_ deleteOldBlund sihs
-
-        when (ccpEpochIndex nextCcp < endEpoch) $
-            loop nextCcp
-
-deleteOldBlund :: ConsolidateM m => SlotIndexHash -> m ()
+-- | A local (to this module) type to control the behaviour of the consolidation
+-- process. A problem faced by this code is that there is no way for it to
+-- figure out the current synchronisation status of the block chain because
+-- that all happens at a higher level and monads/state that this code does not
+-- have access to.
+--
+-- This code also needs to deal with a couple of scenarios on startup:
+--  * This node is syncing the blockchain from scratch and needs to consolidate
+--    older epochs as they arrive.
+--  * This node is fully synced, but has been running old code and no
+--    consolidation has ever been performed.
+--  * This node is running code that does consolidate blocks and all but the
+--    last two blocks have already been consolidated.
+-- During any of the above secnarios, we may also need to deal with a flakey
+-- network connection
+--
+-- To handle all three cases without any synchonisaion feedback, the best we can
+-- do is poll the current tip epoch and compare it to our check point epoch, with
+-- an increase in sleep times between polling. When the sleep time gets over a
+-- certain amount we switch to a mode where this polling is done once every day.
+-- When we are polling more frequently that once a day, if a epoch needs to be
+-- consolidated, the sleep time resets to the minimum (10 seconds).
+data ConsolidateStatus
+    = CSSyncSeconds !Word
+    -- ^ Consolodation is more than two blocks behind and needs to catch up.
+    -- The 'Word' carries the number of seconds to sleep between consolidations.
+    | CSFollowing
+    -- ^ Consolodation is at most two blocks behind the tip so we poll once a
+    -- a day.
+
+statusToSeconds :: ConsolidateStatus -> Integer
+statusToSeconds cs =
+    case cs of
+        CSSyncSeconds s -> fromIntegral s
+        CSFollowing     -> 24 * 60 * 60 -- A whole day
+
+-- Reset the sleep time to the minimum of 2 seconds. When syncing  a large
+-- number of epochs, with a large backlog of unconsolidated block, a fast
+-- network and a slow disk can result epochs being consolidated slower than
+-- they are retrieved from the network.
+resetSyncSeconds :: ConsolidateStatus -> ConsolidateStatus
+resetSyncSeconds st =
+    case st of
+        CSSyncSeconds _ -> CSSyncSeconds 2
```

Comment:
Likewise, I'd prefer to see `minimumSyncSeconds = 2` instead of the magic number, and then Haddock comments can link to it.